examples/5_secret_storage/secret_storage.go

```diff
@@ -0,0 +1,535 @@
+
+	ok, res, err := rental.Remove(team, tool)
+	fmt.Printf("REMOVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	var tools []string
+	tools, err = rental.ListTools(team)
+	fmt.Printf("LIST TOOLS: %+v, %v\n", tools, err)
+
+	res, err = rental.Lookup(team, tool)
+	fmt.Printf("LOOKUP: %+v, %v\n", res, err)
+
+	ok, res, err = rental.Add(team, "time travel machine")
+	fmt.Printf("ADD: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Remove(team, tool)
+	fmt.Printf("REMOVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Add(team, tool)
+	fmt.Printf("ADD: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Reserve(team, user1, tool, date1)
+	fmt.Printf("RESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Reserve(team, user1, tool, date1)
+	fmt.Printf("EXPECTING RESERVE FAIL: %v, %+v, %v\n", ok, res, err)
+	if ok && err == nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Reserve(team, user2, tool, date2)
+	fmt.Printf("RESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	res, err = rental.Lookup(team, tool)
+	fmt.Printf("LOOKUP: %+v, %v\n", res, err)
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date3)
+	fmt.Printf("UNRESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date2)
+	fmt.Printf("EXPECTING UNRESERVE FAIL: %v, %+v, %v\n", ok, res, err)
+	if ok && err == nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date1)
+	fmt.Printf("UNRESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	res, err = rental.Lookup(team, tool)
+	fmt.Printf("LOOKUP: %+v, %v\n", res, err)
+}
+
+func concurrentRentalUser(rental RentalBotClient, team string, tool string, userID int, wg *sync.WaitGroup) {
+	date := fmt.Sprintf("2044-10-0%d", userID)
+	user := fmt.Sprintf("user%d", userID)
+
+	i := 0
+	// keep trying to reserve for user's unique date until successful
+	for {
+		ok, res, err := rental.Reserve(team, user, tool, date)
+		i++
+		fmt.Printf("%v, attempt %d, TRY TO RESERVE: %v, %+v, %+v\n", user, i, ok, res, err)
+		if ok && err == nil {
+			break
+		}
+	}
+	wg.Done()
+}
+
+func concurrentRentalUsers(rental RentalBotClient, team string) {
+	tool := "time travel machine"
+
+	var wg sync.WaitGroup
```

Comment:
https://godoc.org/golang.org/x/sync/errgroup would make this more readable in general